lib/gooddata/rest/rest_aggregator.rb

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+module GoodData
+  module Rest
+    module Aggregator
+      attr_reader :store
+
+      def initialize_store
+        @store = {}
+      end
+
+      def clear_store
+        @store.clear
+      end
+
+      def update_store(domain, method, duration, endpoint)
+        domain = domain.to_sym
+        method = method.to_sym
+        endpoint = endpoint.to_sym
+        @store[domain] = {} unless @store.key?(domain)
+        @store[domain][method] = {} unless @store[domain].key?(method)
+        if @store[domain][method].key?(endpoint)
+          record = @store[domain][method][endpoint]
+          record[:min] = [duration, record[:min]].min
```

Comment:
info: Refers to 'record' more than self (maybe move it to another class?) (FeatureEnvy) (reek)